app/src/main/java/tech/pauly/findapet/shared/LocationHelper.kt

```diff
@@ -0,0 +1,132 @@
+package tech.pauly.findapet.shared
+
+import android.annotation.SuppressLint
+
+import android.content.Context
+import android.location.*
+import android.os.Build
+import android.os.Bundle
+import android.support.annotation.VisibleForTesting
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import tech.pauly.findapet.BuildConfig
+import tech.pauly.findapet.data.ObservableHelper
+import tech.pauly.findapet.data.models.Contact
+import tech.pauly.findapet.dependencyinjection.ForApplication
+import java.io.IOException
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val METERS_TO_MILES = 0.000621371192
+
+@Singleton
+open class LocationHelper @Inject
```

Comment:
Pretty happy with this transition overall. "Helper" is the class that holds the business logic and makes calls to the "Wrapper" which wraps the native APIs so we can mock it.